<?xml version="1.0" ?>
<launch>

    <node pkg="tf" type="static_transform_publisher" name="tf_001" args="0 0 0 0 0 0  map world 10" />

    <arg name="param_file" default="$(find usar_exploration)/config/exploration.yaml"/>
    <rosparam command="load" file="$(arg param_file)"/>
    <arg name="resolution" default="0.1" />
    <arg name="pcl_topic" default="/current_viewpoint_pointcloud" />
    <arg name="voxels_per_side" default="16" />

    <node name="exploration_node" pkg="usar_exploration" type="exploration_node" output="screen">
      
      <!-- settings for voxblox -->
      <!-- Parameters exmplained here: https://voxblox.readthedocs.io/en/latest/pages/The-Voxblox-Node.html#general-parameters-->
      <remap from="voxblox_node/esdf_map_out" to="esdf_map" />
      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
      <param name="publish_esdf_map" value="true" />
      <param name="publish_pointclouds" value="true" />
      <param name="tsdf_voxel_size" value="$(arg resolution)" />
      <param name="method" value="simple" /> 
      <param name="color_mode" value="normals" />
      <param name="max_ray_length_m" value="10.0" />
      <param name="update_mesh_every_n_sec" value="1.0" />
      <param name="use_tf_transforms" value="true" />
      <param name="voxel_carving_enabled" value="true" />
      <param name="min_time_between_msgs_sec" value="0.10" />
      <param name="verbose" value="false" />
      <param name="enable_icp" value="true" />
      <remap from="pointcloud" to="$(arg pcl_topic)"/>

    </node>

   <node name="rviz" pkg="rviz" type="rviz" args=" -d $(find usar_exploration)/rviz/exploration.rviz"/>
   <!--node name="current_view_extraction" pkg="usar_exploration" type="current_view_extraction" output="screen"/-->
</launch>

